shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -48,6 +48,12 @@
   if (!CGSizeEqualToSize(drawable_size, layer.drawableSize)) {
     layer.drawableSize = drawable_size;
   }
+
+  // When there are platform views in the scene, the drawable needs to be presented in the same
+  // transaction as the one created for platform views. When the drawable are being presented from
+  // the raster thread, there is no such transaction.
+  layer.presentsWithTransaction = [[NSThread currentThread] isMainThread];
```

Comment:
In a separate patch perhaps, we should put this in the base implementation of the iOS surface shared by both Skia and Impeller. The models are identical.

Not sure re. testing. We could test that we present with transaction when not on the main thread. But that this is the fix for tearing, I wouldn't know how to test that.